### PR TITLE
Added Registration

### LINKED ISSUE #25: User should be able to register
Needed for #24.

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
fixes #25 
Screenshot after fix:
image
image
image
image

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
@tapasweni-pathak @nikhita @jarifibrahim Please review

File: oshc/.idea/inspectionProfiles/profiles_settings.xml
Comment (jarifibrahim) [inline oshc/.idea/inspectionProfiles/profiles_settings.xml]: This file is generated by Pycharm and you shouldn't add it.

Please remove this file and all the files in the `.idea` directory.

File: oshc/main/static/main/css/app.css
Comment (jarifibrahim) [inline oshc/main/static/main/css/app.css]: Most of the changes in this file are related to indentation. Please undo these indentation changes. It's preferable to use a tab size of 2 spaces in `css` files.

File: oshc/main/templates/base.html
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Indentation error.

File: oshc/main/templates/base.html
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Why change `content` to `container-fluid`?

File: oshc/main/templates/base.html
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Fix indentation.

File: oshc/main/templates/registration/password_reset_confirm.html
Comment (jarifibrahim) [inline oshc/main/templates/registration/password_reset_confirm.html]: Unnecessary new line.

File: oshc/main/templates/registration/password_reset_confirm.html
Comment (jarifibrahim) [inline oshc/main/templates/registration/password_reset_confirm.html]: Unnecessary new line.

File: oshc/oshc/settings.py
Comment (jarifibrahim) [inline oshc/oshc/settings.py]: `DEBUG` should be set to `False`.

File: oshc/oshc/settings.py
Comment (jarifibrahim) [inline oshc/oshc/settings.py]: Change this and following lines to `os.getEnv('EMAIL_USE_TLS)`. This way it is easier to change the email server and related settings.

File: oshc/main/templates/base.html
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Indentation error in the entire file.

File: oshc/main/templates/registration/password_reset_confirm.html
Comment (jarifibrahim) [inline oshc/main/templates/registration/password_reset_confirm.html]: Unnecessary new line.

File: oshc/oshc/settings.py
Comment (jarifibrahim) [inline oshc/oshc/settings.py]: `DEBUG` should be set to `False` by default. Please undo your changes.

File: oshc/main/templates/base.html
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Why change `content` to `container-fluid`?

File: oshc/oshc/urls.py
Comment (jarifibrahim) [inline oshc/oshc/urls.py:7]: You're using the [simple backend](https://django-registration-redux.readthedocs.io/en/latest/simple-backend.html) and you've created the files (email related) for the [default backend](https://django-registration-redux.readthedocs.io/en/latest/default-backend.html).  You should include `registration.backends.default.urls`.

**Review by jarifibrahim — CHANGES_REQUESTED** [review]
@vaibhavsingh97 This is a huge pull request. You should never make these many changes in one PR.  Always try to make atomic commits. :)

One more thing you should be careful about is the indentation settings in your IDE. I believe most of the changes in this PR have been made by the auto-format feature of the IDE.

BTW, I pulled your code and ran it locally. The registration and login process doesn't seem to work.
Please fix it. We cannot merge untill it works. 

You just need to make some minor changes then we can merge your code. Good job :)

**Comment by vaibhavsingh97** [comment]
@jarifibrahim Thanks for reviewing. Can you please tell me the error?
Can we move to Slack for some discussion?

**Comment by vaibhavsingh97** [comment]
@jarifibrahim Updated the suggested changes. Please review again

File: oshc/main/static/main/css/app.css
Comment (jarifibrahim) [inline oshc/main/static/main/css/app.css:20]: This change breaks the layout. You might want to remove it.

File: oshc/main/templates/base.html
Comment (jarifibrahim) [inline oshc/main/templates/base.html:69]: Indentation error.

File: oshc/main/static/main/css/app.css
Comment (jarifibrahim) [inline oshc/main/static/main/css/app.css]: Generally, it is a bad idea to apply styles to bootstrap classes. What if `container-fluid` is used somewhere else on the same page? It will automatically get `padding-left: 0px` which is not the intended behavior. You should add your own class/id if you wish to tweak a bootstrap element.

File: oshc/main/templates/registration/registration_form.html
Comment (jarifibrahim) [inline oshc/main/templates/registration/registration_form.html]: Unnecessary new line.

File: oshc/main/templates/base.html
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Change `static " main/js/script.js " ` to `static "main/js/script.js" `

**Review by jarifibrahim — CHANGES_REQUESTED** [review]
The login/registration isn't working. Does it work on your machine?
Also, you might want to squash your commits into a single commit. Ping me on slack if you don't know how to do it.

**Comment by vaibhavsingh97** [comment]
@jarifibrahim
capture1

**Comment by vaibhavsingh97** [comment]
@tapasweni-pathak Please review and merge.
Steps to run this PR:
- `python manage.py migrate`
- `python manage.py collectstatic`
- `python manage.py runserver`

**Comment by nikhita** [comment]
Thanks for the instructions, @vaibhavsingh97! I checked the changes locally and it works perfectly. :raised_hands: 

@jarifibrahim @tapasweni-pathak  Since you reviewed the PR, I'll leave it to you to merge it.

**Review by jarifibrahim — APPROVED** [review]

File: oshc/main/templates/base.html
Comment (jarifibrahim) [reply oshc/main/templates/base.html]: @vaibhavsingh97 Ping

**Comment by jarifibrahim** [comment]
Merging this since we have 2 approvals. Well done @vaibhavsingh97

**Pull request merged**
